modin/engines/ray/pandas_on_ray/frame/partition_manager.py

```diff
@@ -151,35 +128,27 @@ def get_indices(cls, axis, partitions, index_func=None):
 
     @classmethod
     def broadcast_apply(cls, axis, apply_func, left, right, other_name="r"):
-        def mapper(df, others):
-            other = pandas.concat(others, axis=axis ^ 1)
+        def map_func(df, others):
+            other = pandas.concat(ray.get(others), axis=axis ^ 1)
```

Comment:
Why are we blocking on others?

Reply:
We are blocking on `others` because `others` is a list of `ray.ObjectRefs`. Ray doesn't materialize `ray.ObjectRefs` if they are in a list. We have to materialize them ourselves.

Reply:
Ah ok, thanks for letting me know about this weird ray behavior!

Reply:
If we can pass `others` as `*others` to `apply` it should avoid the need to `ray.get`. I think this way it doesn't give Ray enough information for scheduling and in distributed mode can have some heavy performance implications for scheduling and placement.

Reply:
Currently we can't pass `others` as `*others` to `apply` because `apply` has different interface, i.e. `apply(self, func, **kwargs)`). We already have similar logic related to `ray.get` in `deploy_ray_func`. If we want to change the logic, i.e. get rid of explicit `ray.get`, we should perform that as a part of another issue because that may take a long time, I believe.

Reply:
In that case I prefer to revert these changes since they are mostly refactoring anyway and make the right changes in a later PR.

Reply:
Okay, let's do it as a part of another PR. Reverted the changes.

Reply:
Thanks, the main concern is negatively affecting current cluster users because we do not have sufficient scheduling hints to Ray yet. It is a good change to clean up the code but I do not want to make such performance critical changes so close to a release.

Reply:
I see, agree.